src/services/services.ts

```diff
@@ -136,54 +136,50 @@ namespace ts {
         }
 
         private createChildren(sourceFile?: SourceFileLike) {
-            if (this.kind === SyntaxKind.JSDocComment || isJSDocTag(this)) {
+            if (!isNodeKind(this.kind)) {
+                this._children = emptyArray;
+                return;
+            }
+
+            if (isJSDocComentContainingNode(this)) {
                 /** Don't add trivia for "tokens" since this is in a comment. */
                 const children: Node[] = [];
                 this.forEachChild(child => { children.push(child); });
                 this._children = children;
+                return;
             }

```

Comment:
We will now create children normally for inner jsdoc nodes such as `JSDocTypeExpression`.